examples/deadletterqueue/movingMessagesToDLQ.ts

```diff
@@ -0,0 +1,192 @@
+}
+
+// Helper to process messages by delegating actual processing to a supplied custom messageProcessor
+async function processMessageQueue(messageProcessor: Function, path: string): Promise<void> {
+  const client = ns.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+  await messageProcessor(client);
+}
+
+// OnError handler for message receive
+const onError: OnError = (err: MessagingError | Error) => {
+  console.log(">>>>> Error occurred: ", err);
+};
+
+// OnMessage handler for Scenario A - immediate dead lettering of message on receive
+async function simpleDeadletterProcessor(client: QueueClient): Promise<void> {
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(">>>>> Killing the message", brokeredMessage);
+    await brokeredMessage.deadLetter();
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, { autoComplete: false });
+  await delay(receiveClientTimeout);
+  await receiverHandler.stop();
+}
+
+// OnMessage handler for Scenario B - attempts to exceed max allowed retries
+async function exceedMaxRetriesProcessor(client: QueueClient): Promise<void> {
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(
+      ">>>>> Attempt to abandon message. Current delivery count: ",
+      brokeredMessage.deliveryCount
+    );
+    await brokeredMessage.abandon();
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, { autoComplete: false });
+  await delay(receiveClientTimeout);
+  await receiverHandler.stop();
+}
+
+// OnMessage handler for Scenario C - usage for handling reject/fail on message processing
+async function faultyProcessor(client: QueueClient): Promise<void> {
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(">>>>> Rejecting the message: ", brokeredMessage);
+    try {
+      await Promise.reject("Green apples are not OK.");
+    } catch (err) {
+      console.log("Message from faulty processor: ", err);
+
+      // TODO: Fix bug with .deadletter() in SDK for not setting these properties on the message.
+      brokeredMessage.deadLetter({
+        deadletterReason: "Bad Apple",
+        deadLetterErrorDescription: "Red apples are sweeter than green."
+      });
+    }
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, {
+    autoComplete: false
+  });
+  await delay(receiveClientTimeout);
+  await receiverHandler.stop();
+}
+
+// OnMessage handler for accomplishing queue purge
+async function queuePurgingProcessor(client: QueueClient): Promise<void> {
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    await brokeredMessage.complete();
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, { autoComplete: false });
+  await delay(receiveClientTimeout);
+  await receiverHandler.stop();
+}
+
+main();
```

Comment:
We should use the same pattern as the other samples for this.

Also, as main() returns a `Promise` we need to make sure that we get the result of the promise -- look at the other samples for an example of how to do that.

Reply:
Agree, updated to maintain uniformity.

Reply:
Discussed offline, updated code to handle promise returned on main.